### PR TITLE
Reformat and delint arguments.cc and arguments.hh

### PR DESCRIPTION
**Pull request by omoerbeek**
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/arguments.cc [hunk 1]
```diff
@@ -35,126 +35,124 @@
...
 
   stringtok(parts, param->second, ", \t");
-  for (const auto& part: parts) {
-    if (part == val) {
-      return true;
-    }
-  }
-
-  return false;
+  return std::any_of(parts.begin(), parts.end(), [&](const std::string& str) { return str == val; });
```
Comment (fredmorcos) [inline pdns/arguments.cc:119]: This was available as an LSP/eglot push-button action, right?

**Review by fredmorcos — APPROVED** [review]
I'm fine with the nolints here. The `readdir` rabbit hole is annoying though.

File: pdns/arguments.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/arguments.cc:119]: yes, iirc

File: pdns/arguments.cc [hunk 2]
```diff
@@ -190,30 +192,34 @@ string ArgvMap::configstring(bool running, bool full)
...
   help += formatOne(running, full, "ignore-unknown-settings", helpmap["ignore-unknown-settings"], defaultmap["ignore-unknown-settings"], d_params["ignore-unknown-settings"]);
 
-  for (const auto& i : helpmap) {
-    if (d_typeMap[i.first] == "Command")
+  for (const auto& helpietm : helpmap) {
+    if (d_typeMap[helpietm.first] == "Command") {
       continue;
-    if (i.first == "ignore-unknown-settings")
+    }
+    if (helpietm.first == "ignore-unknown-settings") {
```
Comment (Habbie) [inline pdns/arguments.cc:209]: maybe fix this typo while you're here :)

**Review by Habbie — CHANGES_REQUESTED** [review]
generally approved. One nit that was there already but while you're here, maybe fix it.

I think you can drop the last two commits, together, with no change to PR outcome?

**Review by Habbie — APPROVED** [review]

**Pull request merged**